File: app/src/main/java/com/alexandr7035/txnotes/viewmodel/CreateNoteViewModel.java
package com.alexandr7035.txnotes.viewmodel;

import android.app.Application;

import androidx.lifecycle.ViewModel;

import com.alexandr7035.txnotes.db.NoteEntity;

import java.util.concurrent.ExecutionException;

public class CreateNoteViewModel extends ViewModel {

    private NotesRepository repository;

    public CreateNoteViewModel(Application application) {
        repository = new NotesRepository(application);
    }

    public NoteEntity getNote(int id) throws ExecutionException, InterruptedException {
        return repository.getNoteFromDb(id);
    }

    public void createNote(NoteEntity note) {
        repository.addNoteToDb(note);
    }

}

File: app/src/main/java/by/alexandr7035/txnotes/presentation/create_note/CreateNoteViewModel.kt
package by.alexandr7035.txnotes.presentation.create_note

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import by.alexandr7035.domain.model.CreateNoteModel
import by.alexandr7035.domain.model.Note
import by.alexandr7035.domain.usecase.CreateNoteUseCase
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class CreateNoteViewModel @Inject constructor(private val createNoteUseCase: CreateNoteUseCase): ViewModel() {
    fun saveNote(note: CreateNoteModel) {
        viewModelScope.launch(Dispatchers.IO) {
            createNoteUseCase.execute(note)
        }
    }
}